object-store/src/main/java/org/hypertrace/config/objectstore/ScopeResolutionIdentifiedObjectStoreWithFilter.java

```diff
@@ -0,0 +1,109 @@
+package org.hypertrace.config.objectstore;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Message;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.hypertrace.config.service.change.event.api.ConfigChangeEventGenerator;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+
+/**
+ * ScopeResolutionIdentifiedObjectStoreWithFilter is an abstraction over {@link
+ * IdentifiedObjectStoreWithFilter} to allow scope resolution of fetched configs
+ *
+ * @param <T> Config Object
+ * @param <F> Filter
+ * @param <S> Scope
+ */
+public abstract class ScopeResolutionIdentifiedObjectStoreWithFilter<T extends Message, F, S>
+    extends IdentifiedObjectStoreWithFilter<T, F> {
+
+  protected ScopeResolutionIdentifiedObjectStoreWithFilter(
+      ConfigServiceGrpc.ConfigServiceBlockingStub configServiceBlockingStub,
+      String resourceNamespace,
+      String resourceName,
+      ConfigChangeEventGenerator configChangeEventGenerator) {
+    super(configServiceBlockingStub, resourceNamespace, resourceName, configChangeEventGenerator);
+  }
+
+  public List<T> getAllFilteredResolvedConfigs(RequestContext context, F filter, T defaultConfig) {
+    Map<S, T> configObjectsMap =
+        getAllFilteredConfigs(context, filter).stream()
+            .collect(Collectors.toMap(this::extractScope, Function.identity()));
+
+    return configObjectsMap.entrySet().stream()
+        .map(
+            entry -> resolveConfig(extractScope(entry.getValue()), defaultConfig, configObjectsMap))
+        .collect(Collectors.toUnmodifiableList());
+  }
+
+  public T getFilteredResolvedConfig(RequestContext context, F filter, S scope, T defaultConfig) {
```

Comment:
Where does filter come in here? I think this comes back to using the filtered store as the base, but a call would typically have just a context and scope for the resolved version I'd expect. The scope acts as a filter.

Reply:
umm no - as explained in above comment, 1 config object in a scope can have multiple configs and filter can be something other than scope too..